python/cudf/cudf/core/dataframe.py

```diff
@@ -1499,14 +1500,25 @@ def reset_index(self, drop=False, inplace=False):
             return out.set_index(RangeIndex(len(self)))
 
     def take(self, positions, ignore_index=False):
+        positions = as_column(positions)
+        if pd.api.types.is_bool_dtype(positions):
+            return self._apply_boolean_mask(
+                positions, ignore_index=ignore_index
+            )
+        out = self.gather(positions)
+        if ignore_index:
```

Comment:
Let's just get rid of this kwarg. Pandas doesn't have it and we basically never want it to be `True`. If someone wants this behavior they can call `reset_index` before the `take` call.

Reply:
Done